### ISSUE TITLE
46.beta - The GObject property St.ScrollView.vscroll is deprecated

### ISSUE THREAD

**Issue by tekstryder** [home-sweet-gnome/dash-to-panel#2043]
- Arch Linux | Kernel 6.7.6
- Gnome-shell | Mutter 46.beta
- Wayland (`gnome-shell --no-x11`)
- Dash-to-Panel Master+https://github.com/home-sweet-gnome/dash-to-panel/pull/2028

```
Feb 27 08:54:09 gnome-shell[2298]: The GObject property St.ScrollView.vscroll is deprecated.
                                   0 ensureActorVisibleInScrollView() ["file:///home/myusername/.local/share/gnome-shell/extensions/[dash-to-panel@jderose9.github.com/utils.js](http://dash-to-panel@jderose9.github.com/utils.js)":508:22]
                                   1 _createAppItem/</<() ["file:///home/myusername/.local/share/gnome-shell/extensions/[dash-to-panel@jderose9.github.com/taskbar.js](http://dash-to-panel@jderose9.github.com/taskbar.js)":733:32]
                                   2 _create/this[name]<() ["file:///home/myusername/.local/share/gnome-shell/extensions/[dash-to-panel@jderose9.github.com/utils.js](http://dash-to-panel@jderose9.github.com/utils.js)":167:12]
                                   3 anonymous() ["resource:///org/gnome/shell/ui/init.js":21:19]
Feb 27 08:54:09 gnome-shell[2298]: The GObject property St.ScrollView.hscroll is deprecated.
                                   0 ensureActorVisibleInScrollView() ["file:///home/myusername/.local/share/gnome-shell/extensions/[dash-to-panel@jderose9.github.com/utils.js](http://dash-to-panel@jderose9.github.com/utils.js)":509:22]
                                   1 _createAppItem/</<() ["file:///home/myusername/.local/share/gnome-shell/extensions/[dash-to-panel@jderose9.github.com/taskbar.js](http://dash-to-panel@jderose9.github.com/taskbar.js)":733:32]
                                   2 _create/this[name]<() ["file:///home/myusername/.local/share/gnome-shell/extensions/[dash-to-panel@jderose9.github.com/utils.js](http://dash-to-panel@jderose9.github.com/utils.js)":167:12]
                                   3 anonymous() ["resource:///org/gnome/shell/ui/init.js":21:19]
```

**Comment by darkblaze69** [comment]
Can be fixed like here https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/998d703bc24441dae5f1ac5436de15ab7527b7eb

I have it fixed for myself like this, not sure if is all 100% correct, but no warning and visible issues:

<details><summary>hv-deprecations.patch:</summary>
<p>

```
diff --git a/taskbar.js b/taskbar.js
index e1f3e0f..8b599bd 100644
--- a/taskbar.js
+++ b/taskbar.js
@@ -142,7 +142,7 @@ export const TaskbarActor = GObject.registerClass({
 
         scrollview.allocate(childBox);
 
-        let [value, , upper, , , pageSize] = scrollview[orientation[0] + 'scroll'].adjustment.get_values();
+        let [value, , upper, , , pageSize] = scrollview[orientation[0] + 'adjustment'].get_values();
         upper = Math.floor(upper);
         scrollview._dtpFadeSize = upper > pageSize ? this._delegate.iconSize : 0;
 
@@ -277,7 +277,7 @@ export const Taskbar = class extends EventEmitter {
             x_align: rtl ? Clutter.ActorAlign.END : Clutter.ActorAlign.START
         });
 
-        let adjustment = this._scrollView[orientation[0] + 'scroll'].adjustment;
+        const adjustment = this._scrollView[orientation[0] + 'adjustment'];
         
         this._workId = Main.initializeDeferredWork(this._box, this._redisplay.bind(this));
 
@@ -514,7 +514,7 @@ export const Taskbar = class extends EventEmitter {
 
         let adjustment, delta;
 
-        adjustment = this._scrollView[orientation[0] + 'scroll'].get_adjustment();
+        adjustment = this._scrollView[orientation[0] + 'adjustment'];
 
         let increment = adjustment.step_increment;


diff --git a/windowPreview.js b/windowPreview.js
index 3ab0730..f0a2741 100644
--- a/windowPreview.js
+++ b/windowPreview.js
@@ -509,7 +509,7 @@ export const PreviewMenu = GObject.registerClass({
     }
 
     _getScrollAdjustmentValues() {
-        let [value , , upper, , , pageSize] = this._scrollView[(this.isVertical ? 'v' : 'h') + 'scroll'].adjustment.get_values();
+        let [value , , upper, , , pageSize] = this._scrollView[(this.isVertical ? 'v' : 'h') + 'adjustment'].get_values();
 
         return [value, upper, pageSize];
     }

diff --git a/utils.js b/utils.js
index 8fa2986..828e184 100644
--- a/utils.js
+++ b/utils.js
@@ -505,8 +505,8 @@ export const notify = function(text, iconName, action, isTransient) {
  * Return the amount of shift applied
 */
 export const ensureActorVisibleInScrollView = function(scrollView, actor, fadeSize, onComplete) {
-    let vadjustment = scrollView.vscroll.adjustment;
-    let hadjustment = scrollView.hscroll.adjustment;
+    const vadjustment = scrollView.vadjustment;
+    const hadjustment = scrollView.hadjustment;
     let [vvalue, vlower, vupper, vstepIncrement, vpageIncrement, vpageSize] = vadjustment.get_values();
     let [hvalue, hlower, hupper, hstepIncrement, hpageIncrement, hpageSize] = hadjustment.get_values();

```
</p>
</details>

**Comment by tekstryder** [comment]
> I have it fixed for myself like this, not sure if is all 100% correct, but no warning and visible issues

Been using your patch for a few days without issue. Looks correct based on shell commit. 

Thanks for saving me time here!

**Comment by philippun1** [comment]
@darkblaze69 I added your patch to #2028

**Comment by tekstryder** [comment]
Verified fixed in v61 tag on Gnome 46.0.

Closing
